### PR TITLE
Only add size of non whitelisted files and directories

### PR DESCRIPTION
**Pull request by Refresh06**

### CODE REVIEW

File: src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java [hunk 1]
```diff
@@ -29,24 +29,44 @@ public DeleteAction(String... paths) {
...
+                    LOG.error(e.getMessage(), e);
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+
+        return size;
+    }
+
+    private long sizeOfDirectory(File directory) {
```
Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java]: Isn't this what the `FileUtils.sizeOfDirectory(versionDir);` does already? 

Ref - https://github.com/apache/commons-io/blob/master/src/main/java/org/apache/commons/io/FileUtils.java#L2513-L2528

Comment (ashwanthkumar) [reply src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java]: Oops, I missed the `isNotWhiteListed` part. Sorry.

File: src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java [hunk 2]
```diff
@@ -29,24 +29,44 @@ public DeleteAction(String... paths) {
...
-        } else {
-            LOG.info("Deleting " + versionDir.getAbsolutePath() + ", size = " + FileUtils.byteCountToDisplaySize(size));
-            try {
-                deleteDirectory(versionDir);
-            } catch (IOException e) {
-                LOG.error(e.getMessage(), e);
-                throw new RuntimeException(e);
+        long size = sizeOfDirectory(versionDir);
+
+        if (size > 0) {
```
Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java]: Do we really need this (`size > 0`) check? Isn't the operation no-op if the size is 0?

File: src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java [hunk 3]
```diff
@@ -29,24 +29,44 @@ public DeleteAction(String... paths) {
...
+            LOG.info((dryRun ? "[DRY RUN] Will remove " : "Deleting ") + versionDir.getAbsolutePath() + ", size = " + FileUtils.byteCountToDisplaySize(size));
```
Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java]: Not a big fan of using ternary conditions while doing string concat. Can we please have simple if-else here?

**Comment by ashwanthkumar** [comment]
@Refresh06 I've added a few comments, please do have a look and let me know your thoughts.

File: src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java (hunk 2 above)
Comment (Refresh06) [reply src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java]: If the folder has already been cleaned up, then all files are whitelisted and the size will be 0. Without this check we log that we are cleaning up 0 bytes in a lot of folders.

File: src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java (hunk 3 above)
Comment (Refresh06) [reply src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java]: Sure thing, will update

File: src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java (hunk 2 above)
Comment (Refresh06) [reply src/main/java/in/ashwanthkumar/gocd/actions/DeleteAction.java]: A failing test reveals that this check leaves 0 size files in the folders (not a big issue, but messy).
Will fix.

**Comment by Refresh06** [comment]
@ashwanthkumar I've updated based on your comments and the failing testcase. Please have a look.

**Comment by ashwanthkumar** [comment]
Awesome, thanks @Refresh06. Merging this.

**Pull request merged**